api/v1alpha1/backend_types.go

```diff
@@ -113,6 +113,13 @@ type UnixSocket struct {
 
 // BackendSpec describes the desired state of BackendSpec.
 type BackendSpec struct {
+	// Type defines the type of the backend. Defaults to "Endpoints"
+	//
+	// +kubebuilder:validation:Enum=Endpoints;DynamicForwardProxy
+	// +kubebuilder:default=Endpoints
+	// +optional
+	Type BackendType `json:"type"`
```

Comment:
if this's optional, it should be an point and with `omitempty `.

Reply:
so not to introduce the breaking change, this needs to be optional. However, we haven't used a pointer for "type" elsewhere (i guess this will be the first time?). is that ok with you

Reply:
done